promql/prettier/prettier.go

```diff
@@ -0,0 +1,499 @@
+					formatItems = true
+				}
+			}
+			if formatItems {
+				var (
+					tempItems              []parser.Item
+					finishedKeywordWriting bool
+				)
+				// Get index of the closing paren.
+				for j := keywordIndex; j <= len(items); j++ {
+					if items[j].Typ == parser.RIGHT_PAREN {
+						closingParenIndex = j
+						break
+					}
+				}
+				if closingParenIndex == itemNotFound {
+					panic("invalid paren index: closing paren not found")
+				}
+				// Re-order lexical items. TODO: consider using slicing of lists.
+				for itemp := 0; itemp < len(items); itemp++ {
+					if itemp <= aggregatorIndex || itemp > closingParenIndex {
+						tempItems = append(tempItems, items[itemp])
+					} else if !finishedKeywordWriting {
+						// Immediately after writing the aggregator, write the keyword and
+						// the following expression till the closing paren. This is done
+						// to be in-order with the expected result.
+						for j := keywordIndex; j <= closingParenIndex; j++ {
+							tempItems = append(tempItems, items[j])
+						}
+						// itemp has not been updated yet and its holding the left paren.
+						tempItems = append(tempItems, items[itemp])
+						finishedKeywordWriting = true
+					} else if !(itemp >= keywordIndex && itemp <= closingParenIndex) {
+						tempItems = append(tempItems, items[itemp])
+					}
+				}
+				items = tempItems
+			}
+
+		case parser.IDENTIFIER:
+			itr := advanceComments(items, i)
+			if i < 1 || item.Val != "__name__" || items[itr+2].Typ != parser.STRING {
+				continue
+			}
+			var (
+				leftBraceIndex  = itemNotFound
+				rightBraceIndex = itemNotFound
+				labelValItem    = items[itr+2]
+				metricName      = labelValItem.Val[1 : len(labelValItem.Val)-1] // Trim inverted-commas.
+				tmp             []parser.Item
+				skipBraces      bool // For handling metric_name{} -> metric_name.
+			)
+			if isMetricNameNotAtomic(metricName) {
+				continue
+			}
+			for backScanIndex := itr; backScanIndex >= 0; backScanIndex-- {
+				if items[backScanIndex].Typ == parser.LEFT_BRACE {
+					leftBraceIndex = backScanIndex
+					break
+				}
+			}
+			for forwardScanIndex := itr; forwardScanIndex < len(items); forwardScanIndex++ {
+				if items[forwardScanIndex].Typ == parser.RIGHT_BRACE {
+					rightBraceIndex = forwardScanIndex
+					break
+				}
+			}
+			if leftBraceIndex == itemNotFound || rightBraceIndex == itemNotFound {
+				continue
+			}
+			itr = advanceComments(items, itr)
+			if items[itr+3].Typ == parser.COMMA {
+				skipBraces = rightBraceIndex-5 == advanceComments(items, leftBraceIndex)
+			} else {
+				skipBraces = rightBraceIndex-4 == advanceComments(items, leftBraceIndex)
+			}
+			identifierItem := parser.Item{Typ: parser.IDENTIFIER, Val: metricName, Pos: 0}
+			// The code below re-orders the lex items in the items array and update the same. The re-ordering of
+			// lex items should be such that the key of any key-value pair if is `__name__` and its corresponding value
```

Comment:
This doesn't parse

Reply:
I didn't get the point. Could you please explain?

Reply:
This sentence makes no sense grammatically.

Reply:
Updated the explanation. Please give it a read and comment on its satisfactoriness.